### PR TITLE
implemented subscription checkout and fixed error re plan intervals

### PR DESCRIPTION
**Pull request by ghost**
-added plan_name to the database and migrated;
-added carted_subscription to the charges controller;
-added plan_name into carted products controller;
-added plan_name into carted_subscriptions react component and removed plan.id; and
-fixed different plan interval issue with stripe not allowing different intervals of subscriptions to plans for the same product.

### CODE REVIEW

File: app/controllers/carted_products_controller.rb [hunk 1]
```diff
@@ -33,18 +33,34 @@ def create
...
+        carted_subscriptions = CartedSubscription.where(
+          status: 'carted',
+          customer_id: guest_or_customer_id
+        )
+
         plans = Stripe::Plan.list(limit: 50)
         plan = StripeTool.find_plan(plans, params[:plan_id], params[:product_id])
+
+        carted_subscriptions.each do |cs|
+          if cs.interval_count != plan.first.interval_count || cs.interval != plan.first.interval
```
[the remark this answers is not in the document]
Comment (ghost) [reply app/controllers/carted_products_controller.rb]: ?? - I'm confused on this error.

[the remark this answers is not in the document]
Comment (heydanhey) [reply app/controllers/carted_products_controller.rb]: [ruby next statement](https://stackoverflow.com/questions/2278285/next-versus-if-in-a-each-loop)

**Review by heydanhey — APPROVED** [review]
+1, Looks good. Is it actually working?

**Comment by ghost** [comment]
Yeah, I went onto the Stripe dashboard and confirmed that my plan went
through, and it also updated on my customer dashboard.

We will need to route it somewhere, either to invoice page or back to
customer dashboard so they can see they have been subscribed. I created
trello cards for that and the shipping charge that we will need to
integrate still.

**Comment by heydanhey** [comment]
hurray

**Pull request merged**